plugins/inputs/http_listener_v2/http_listener_v2_test.go

```diff
@@ -724,6 +727,69 @@ func TestServerHeaders(t *testing.T) {
 	require.Equal(t, "value", resp.Header.Get("key"))
 }
 
+func TestUnixSocket(t *testing.T) {
+	listener, err := newTestHTTPListenerV2()
+	require.NoError(t, err)
+	file, err := os.CreateTemp("", "*.socket")
+	require.NoError(t, err)
+	require.NoError(t, file.Close())
+	defer os.Remove(file.Name())
+	socketName := file.Name()
+	listener.ServiceAddress = "unix://" + socketName
+	listener.SocketMode = "777"
+	acc := &testutil.Accumulator{}
+	require.NoError(t, listener.Init())
+	require.NoError(t, listener.Start(acc))
+	defer listener.Stop()
+	httpc := http.Client{
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", socketName)
+			},
+		},
+	}
+	resp, err := httpc.Post(createURL(listener, "http", "/write", "db=mydb"), "", bytes.NewBufferString(testMsg))
+	require.NoError(t, err)
+	require.NoError(t, resp.Body.Close())
+	require.EqualValues(t, 204, resp.StatusCode)
+}
+
+func TestServiceAddressURL(t *testing.T) {
+	unixSocket := filepath.FromSlash(os.TempDir() + "/test.sock")
+	cases := []struct {
+		serviceAddress, expectedAddress string
+		expectedPort                    int
+		shouldError                     bool
+	}{
+		{":8080", "[::]:8080", 8080, false},
+		{"localhost:4123", "127.0.0.1:4123", 4123, false},
+		{"tcp://localhost:4321", "127.0.0.1:4321", 4321, false},
+		{"127.0.0.1:9443", "127.0.0.1:9443", 9443, false},
+		{"tcp://127.0.0.1:8443", "127.0.0.1:8443", 8443, false},
+		{"tcp://:8443", "[::]:8443", 8443, false},
+		// port not provided
+		{"8.8.8.8", "", 0, true},
+		{"unix://" + unixSocket, unixSocket, 0, false},
+		// wrong protocol
+		{"notexistent:///tmp/test.sock", "", 0, true},
+	}
+	for _, c := range cases {
+		listener, err := newTestHTTPListenerV2()
+		require.NoError(t, err)
+		listener.ServiceAddress = c.serviceAddress
+		err = listener.Init()
+		require.Equal(t, c.shouldError, err != nil, "Init returned wrong error result error is: %q", err)
+		if c.shouldError {
+			continue
+		}
+		acc := &testutil.Accumulator{}
+		require.NoError(t, listener.Start(acc))
+		require.Equal(t, c.expectedAddress, listener.listener.Addr().String())
+		require.Equal(t, c.expectedPort, listener.Port)
+		listener.Stop()
+	}
+}
+
```

Comment:
This test is problematic in CI environments. Imagine another test using the same port, if they by chance run at the same time one of them will fail and we get flaky tests. I would remove the test here or solely stick to the URL parsing part without actually "starting" the listener...

Reply:
Yes I agree such test might be problematic when running in parallel with others, but afaik they are run synchronously and there are also other tests that actually, start listening on port. 
Anyway my logic should be tested enough without `Start` as it is located in `Init` function so I will remove the `Start` method call.

Reply:
Well actually the `Init` already calls the `net.Listen` so that the port gets busy and we need to stop the internal `dial.Listener`. 

I think my changes are big enough especially with special windows case that having unit test for url parsing would be nice so please reconsider the idea of fully removing test, but if you insist I will remove.

Reply:
Well starting the listener in `Init()` is also wrong, this should be done in `Start()`... Anyhow, as you didn't introduce this, I don't blame you. ;-)

Regarding the tests: Yes, the tests in this file/package are executes sequentially, but all packages are executed in parallel (in the best case)... So if e.g. `outputs.http` opens a server on this port we run into trouble... So it is acceptable to specify no port or port zero as this will allow the OS to choose a free port but the above will definitively bite us. Been there suffered that.

In any case, what you do is to test if `url.Parse` works I think and I guess we should assume this works at least as long as this test will open a fixed port... So either change the test/code to not starting listen or remove the fixed ports.

Reply:
Okay lets make it right then I agree that the test can be actually removed especially when windows unix path works fine with url.Parse. 
I will move the `.Listen` part to `Start` then.

Reply:
Done in [45792b5](https://github.com/influxdata/telegraf/pull/15764/commits/45792b54d0196db88386fbf12227f446dea2109f)